### PR TITLE
adds focus view

### PR DESCRIPTION
**Pull request by d13**

### CODE REVIEW

File: src/plus/subscription/serverConnection.ts [hunk 1]
```diff
@@ -12,12 +12,20 @@ import type { DeferredEvent, DeferredEventExecutor } from '../../system/event';
 import { promisifyDeferred } from '../../system/event';
 
 export const AuthenticationUriPathPrefix = 'did-authenticate';
+// TODO: What user-agent should we use?
+const userAgent = 'Visual-Studio-Code-GitLens';
```
Comment (axosoft-ramint) [inline src/plus/subscription/serverConnection.ts:16]: Just noting this TODO here. Looks like it's a pre-existing TODO from before that was moved here. Not sure if necessary to resolve before this is merged.

File: src/plus/webviews/focus/focusWebview.ts [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+		this._pullRequests = allPrs.sort((a, b) => {
+			const scoreA = getScore(a);
+			const scoreB = getScore(b);
+
+			if (scoreA === scoreB) {
+				return a.pullRequest.date.getTime() - b.pullRequest.date.getTime();
+			}
+			return (scoreB ?? 0) - (scoreA ?? 0);
+		});
+		// }
```
Comment (axosoft-ramint) [inline src/plus/webviews/focus/focusWebview.ts:242]: commented line, can be removed

File: src/plus/webviews/focus/focusWebview.ts [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+		return this._pullRequests;
+	}
+
+	private async getMyIssues(richRepos: RepoWithRichRemote[]): Promise<SearchedIssue[]> {
+		// if (this._issues.length === 0) {
```
Comment (axosoft-ramint) [inline src/plus/webviews/focus/focusWebview.ts:248]: commented line, can be removed

File: src/plus/webviews/focus/focusWebview.ts [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+		for (const { remote } of richRepos) {
+			const issues = await this.container.git.getMyIssues(remote);
+			if (issues == null) {
+				continue;
+			}
+			allIssues.push(...issues.filter(pr => pr.reasons.length > 0));
+		}
+
+		this._issues = allIssues.sort((a, b) => b.issue.updatedDate.getTime() - a.issue.updatedDate.getTime());
+		// }
```
Comment (axosoft-ramint) [inline src/plus/webviews/focus/focusWebview.ts:259]: commented line, can be removed

File: src/plus/webviews/focus/focusWebview.ts [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+import type { Subscription } from '../../../subscription';
+import { SubscriptionState } from '../../../subscription';
+import { registerCommand } from '../../../system/command';
+import { WebviewBase } from '../../../webviews/webviewBase';
+import type { SubscriptionChangeEvent } from '../../subscription/subscriptionService';
+import { ensurePlusFeaturesEnabled } from '../../subscription/utils';
+import type { State } from './protocol';
+import { DidChangeStateNotificationType, DidChangeSubscriptionNotificationType } from './protocol';
+
+interface RepoWithRichRemote {
```
Comment (axosoft-ramint) [inline src/plus/webviews/focus/focusWebview.ts:27]: Should this go to the `protocol` file?

**Review by axosoft-ramint — COMMENTED** [review]
Just tiny little things. Did a quick spot-test by pulling open the focus view for my single-repo workspace and it worked fine. Nice work!

File: src/git/models/pullRequest.ts [hunk 6]
```diff
@@ -14,14 +14,61 @@ export const enum PullRequestState {
...
 	readonly state: PullRequestState;
 	readonly mergedDate?: Date;
+	readonly refs?: PullRequestRefs;
+	readonly isDraft?: boolean;
+	readonly additions?: number;
+	readonly deletions?: number;
+	readonly comments?: number;
+	readonly mergeableState?: PullRequestMergeableState;
+	readonly reviewDecision?: PullRequestReviewDecision;
+	readonly reviewRequests?: { isCodeOwner: boolean; reviewer: PullRequestMember }[];
```
[the remark this answers is not in the document]
Comment (d13) [reply src/git/models/pullRequest.ts:65]: Oops, I made the `PullRequestReviewer` for it and didn't update this line. Great catch!

File: src/plus/subscription/serverConnection.ts (hunk 1 above)
Comment (d13) [reply src/plus/subscription/serverConnection.ts:16]: I'll have to follow-up with Eric on that, but its a pretty old TODO from where it came from and not really needed to solve now.

File: src/plus/webviews/focus/focusWebview.ts (hunk 5 above)
Comment (d13) [reply src/plus/webviews/focus/focusWebview.ts:27]: no, this is a local type just for the webview, this shape is serialized into `RepoWithRichProvider` which is in the protocol.

File: src/plus/webviews/focus/focusWebview.ts (hunk 2 above)
Comment (d13) [reply src/plus/webviews/focus/focusWebview.ts:242]: done

File: src/plus/webviews/focus/focusWebview.ts (hunk 3 above)
Comment (d13) [reply src/plus/webviews/focus/focusWebview.ts:248]: done

File: src/plus/webviews/focus/focusWebview.ts (hunk 4 above)
Comment (d13) [reply src/plus/webviews/focus/focusWebview.ts:259]: done

**Review by axosoft-ramint — APPROVED** [review]
Awesome work! Excited to use this every day! 💥

**Pull request merged**